src/main/java/nextstep/courses/domain/session/enrollment/Enrollment.java

```diff
@@ -1,9 +1,54 @@
 package nextstep.courses.domain.session.enrollment;
 
-import nextstep.courses.domain.session.SessionStatus;
 import nextstep.users.domain.NsUser;
 
-public interface Enrollment {
-    void enroll(NsUser user);
-    SessionStatus getStatus();
+public class Enrollment {
+    private final NsUser user;
+    private EnrollmentStatus enrollmentStatus;
+
+    public Enrollment(NsUser user, EnrollmentStatus enrollmentStatus) {
+        this.user = user;
+        this.enrollmentStatus = enrollmentStatus;
+    }
+
+    public Enrollment(NsUser user) {
+        this(user, EnrollmentStatus.PENDING_APPROVAL);
+    }
+
+    public NsUser getUser() {
+        return user;
+    }
+
+    public EnrollmentStatus getEnrollmentStatus() {
+        return enrollmentStatus;
+    }
+
+    public void approve() {
+        if (enrollmentStatus == EnrollmentStatus.PENDING_APPROVAL) {
+            enrollmentStatus = EnrollmentStatus.ENROLLED;
+        } else {
+            throw new IllegalStateException("승인 대기 상태의 수강신청만 승인할 수 있습니다.");
+        }
```

Comment:
```suggestion
        enrollmentStatus = enrollmentStatus.approve();
```
이 로직 구현을 EnrollmentStatus에 구현하는 것은 어떨까?